rs/sns/governance/src/governance.rs

```diff
@@ -5461,13 +5541,17 @@ impl Governance {
                 .clone_from(&target_version.archive_wasm_hash);
         }
 
-        let deployed_version = match self.proto.deployed_version.clone() {
-            None => {
+        let (current_version, new_cached_upgrade_steps) = match self
+            .proto
+            .cached_upgrade_steps_or_err()
+            .and_then(CachedUpgradeSteps::consume)
```

Comment:
> Doesn't this fail if there isn't at least one entry after current? 

Yes

> That is definitely not what we want here

In which case would we expect the cached upgrade steps to have only one version during an SNS upgrade?

Reply:
This was discussed over zoom and has been addressed in the current version I think. In case anyone else reads the comments, the cached upgrade steps may have only one version in various situations, such as when the upgrade steps change between dispatching the upgrade and checking its success, or when dispatching an upgrade through UpgradeSnsToNextVersion proposals